src/System.Private.CoreLib/src/System/StubHelpers.cs

```diff
@@ -1047,7 +1034,7 @@ private unsafe IntPtr ConvertStringBuilderToNative(StringBuilder pManagedHome, i
                 StubHelpers.CheckStringLength(pManagedHome.Capacity);
 
                 // marshal the object as Ansi string (UnmanagedType.LPStr)
-                int allocSize = (pManagedHome.Capacity * Marshal.SystemMaxDBCSCharSize) + 4;
+                int allocSize = checked((pManagedHome.Capacity * Marshal.SystemMaxDBCSCharSize) + 4);
```

Comment:
Am I to assume the `+ 4` is because of `BSTR` semantics? If so can we doc that as well. I don't fully understand why that is true based on the comment above though. What am I missing?

Reply:
> `+ 4` is because of `BSTR` semantics?

Looking at the code that follows, my guess that this is some kind of compat quirk. There was probably some broken .NET Framework app with buffer overrun, and we have made it "work" by allocating bigger buffer than strictly necessary and filling that buffer with 3 extra null terminators. I am just speculating. I was not able to trace down where this came from in the source control history.